### PR TITLE
Add OpenTelemetry parameter constructors to all api Tracer classes

### PR DESCRIPTION
**Pull request by evantorrie**
Add an `OpenTelemetry` parameter to api RPC and DatabaseTracer constructors, to match the same constructor that was added to `BaseTracer`.

Ensure all the constructors call `super(..)` so that they take on the behavior of `BaseTracer` where context propagators and tracer are automatically initialized.

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcClientTracer.java [hunk 1]
```diff
@@ -5,14 +5,28 @@
...
 package io.opentelemetry.instrumentation.api.tracer;
 
+import io.opentelemetry.api.OpenTelemetry;
 import io.opentelemetry.api.trace.Tracer;
 
 public abstract class RpcClientTracer extends BaseTracer {
-  protected RpcClientTracer() {}
+  protected RpcClientTracer() {
+    super();
+  }
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcClientTracer.java:14]:
```suggestion
  protected RpcClientTracer() {}
```

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcServerTracer.java [hunk 2]
```diff
@@ -5,16 +5,28 @@
...
 package io.opentelemetry.instrumentation.api.tracer;
 
+import io.opentelemetry.api.OpenTelemetry;
 import io.opentelemetry.api.trace.Tracer;
 import io.opentelemetry.context.propagation.TextMapPropagator;
 
 public abstract class RpcServerTracer<REQUEST> extends BaseTracer {
 
-  protected RpcServerTracer() {}
+  protected RpcServerTracer() { super(); }
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcServerTracer.java:14]:
```suggestion
  protected RpcServerTracer() {}
```

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java [hunk 3]
```diff
@@ -21,10 +20,12 @@
 
   protected static final String DB_QUERY = "DB Query";
 
-  protected final Tracer tracer;
-
   public DatabaseClientTracer() {
-    tracer = GlobalOpenTelemetry.getTracer(getInstrumentationName(), getVersion());
+    super();
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:24]: if you accept this, spotless will make you collapse the empty constructor to a single line `public DatabaseClientTracer() {}`
```suggestion
```

**Review by jkwatson — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java (hunk 3 above)
Comment (evantorrie) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java]: made updates to placate spotless

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcClientTracer.java (hunk 1 above)
Comment (evantorrie) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/RpcClientTracer.java]: made updates to placate spotless

**Review by trask — APPROVED** [review]
Thanks @evantorrie!

**Pull request merged**